### PR TITLE
Implement custom maps functionality

### PR DESCRIPTION
**Pull request by jpmonettas**

### CODE REVIEW

File: src/cljs/analysis_viewer/events/maps.cljs [hunk 1]
```diff
@@ -25,10 +24,13 @@
...
-                                      (let [world-map? (string/ends-with? map-code "WORLD")]
-                                        [:map/load-map (cond-> {:map/url (str s3-bucket-url "maps/country-code-maps/" map-code ".json")}
-                                                         world-map? (assoc :map/z-index 0))]))))
+                              (mapv (fn [{:keys [kind] :as map}]
+                                      [:map/load-map
+                                       (case kind
+                                         :world          {:map/url (str s3-bucket-url "maps/country-code-maps/WORLD.json") :map/z-index 0}
+                                         :country-detail {:map/url (str s3-bucket-url "maps/country-code-maps/" (:code map) ".json")}
+                                         :custom         {:map/url (str s3-bucket-url (:path map))})])))
+        _ (prn load-maps-events)
```
Comment (fbielejec) [inline src/cljs/analysis_viewer/events/maps.cljs]: leftover?

File: src/cljs/ui/analysis_results/page.cljs [hunk 2]
```diff
@@ -42,11 +44,13 @@
      [discrete-rates analysis]
      nil)])
 
-(defn results [{:keys [bayes-factors] :as analysis}]
+(defn results [{:keys [bayes-factors custom-map] :as analysis}]
   (let [config @(re-frame/subscribe [::subs/config])
         viewer-host (:analysis-viewer-url config)
         {:keys [viewer-url-params]} (:analysis analysis)
-        viewer-url (str viewer-host "/" viewer-url-params)]
+        [_ custom-map-url] (when custom-map (re-find #".+://.+?/.+?/(.+)" (:file-url custom-map)))
```
Comment (fbielejec) [inline src/cljs/ui/analysis_results/page.cljs]: what's the deal here with the regex? Can you leave a comment or refactor to utility function?

File: src/cljs/ui/events/graphql.cljs [hunk 3]
```diff
@@ -153,6 +153,17 @@
...
+  [{:keys [db]} _ {:keys [analysis-id] :as custom-map}]
+  (js/console.log "Custom map uploaded" custom-map)
+  {:db (assoc-in db [:analysis analysis-id :custom-map] custom-map)})
+
+(defmethod handler :delete-custom-map
+  [{:keys [db]} _ analysis-id]
+  (js/console.log "Custom map deleter" analysis-id)
+  {:db (update-in db [:analysis analysis-id] dissoc :custom-map)})
+
+
```
Comment (fbielejec) [inline src/cljs/ui/events/graphql.cljs]: extra newline

File: src/cljs/analysis_viewer/events/maps.cljs (hunk 1 above)
Comment (jpmonettas) [reply src/cljs/analysis_viewer/events/maps.cljs]: yeah, removed

File: src/cljs/ui/analysis_results/page.cljs (hunk 2 above)
Comment (jpmonettas) [reply src/cljs/ui/analysis_results/page.cljs]: yeah, makes sense, done

File: src/cljs/ui/events/graphql.cljs (hunk 3 above)
Comment (jpmonettas) [reply src/cljs/ui/events/graphql.cljs]: done

File: src/cljs/ui/analysis_results/page.cljs (hunk 2 above)
Comment (fbielejec) [reply src/cljs/ui/analysis_results/page.cljs]: uhm,.. I see no changes :)

**Review by fbielejec — APPROVED** [review]
LGTM, good job!

**Pull request merged**